modules/virt-accessing-vmi-ssh.adoc

```diff
@@ -4,51 +4,82 @@
-machine.
+[NOTE]
+====
+Heavy SSH traffic on the control plane can slow down the API server. If you regularly need a high amount of connections and traffic, use a dedicated Kubernetes `Service` object to access the virtual machine.
+====
 
 .Prerequisites
-* You must be in the same project as the VMI.
-* The VMI you want to access must be connected
-to the default pod network by using the `masquerade` binding method.
-* The VMI you want to access must be running.
-* Install the OpenShift CLI (`oc`).
+* You have access to an {product-title} cluster with `cluster-admin` permissions.
+* You have installed the OpenShift CLI (`oc`).
+* You have installed the `virtctl` client.
+* The virtual machine you want to access is running.
+
 
 .Procedure
-. Run the following command to create the `fedora-vm-ssh` service:
+
+. Use the `ssh-keygen` command to generate an SSH public key pair:
 +
+[source,terminal]
 ----
-$ virtctl expose vm <fedora-vm> --port=22 --name=fedora-vm-ssh --type=NodePort <1>
+$ ssh-keygen -f id_rsa
 ----
-<1> `<fedora-vm>` is the name of the VM that you run the
-`fedora-vm-ssh` service on.
 
-. Check the service to find out which port the service acquired:
+. Create an SSH authentication secret which contains the SSH public key to access the VM:
 +
 [source,terminal]
 ----
-$ oc get svc
+$ oc create secret generic my-pub-key --from-file=key1=id_rsa.pub
 ----
+
+. Add a reference to the secret in the `VirtualMachine` manifest. For example:
++
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: testvm
+spec:
+  running: true
+  template:
+    spec:
+      accessCredentials:
+      - sshPublicKey:
+          source:
+            secret:
+              secretName: my-pub-key <1>
+          propagationMethod:
+            configDrive: {} <2>
+# ...
+----
+<1> Reference to the SSH authentication `Secret` object.
+<2> The SSH public key is injected into the VM as metadata using the `configDrive` cloud-init provider.
+
+. Restart the VM to apply your changes.
+
+. Run the following command to access the VM via SSH:
 +
-.Example output
 [source,terminal]
 ----
-NAME            TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)           AGE
-fedora-vm-ssh   NodePort   127.0.0.1      <none>        22:32551/TCP   6s
+$ virtctl ssh -i id_rsa <vm_username>@<vm_name>
 ----
+
+. Optional: To securely transfer files to or from the VM, use the following commands:
 +
-In this example, the service acquired the `32551` port.
 
-. Log in to the VMI via SSH. Use the `ipAddress` of any of the cluster
-nodes and the port that you found in the previous step:
+.Copy a file from your machine to the VM
+[source,terminal]
+----
+$ virtctl scp <filename> <vm_name>
+----
 +
+.Copy a file from the VM to your machine
 [source,terminal]
 ----
-$ ssh username@<node_IP_address> -p 32551
+$ virtctl scp <vm_name> <filename>
```

Comment:
The identity file and user name are needed here too. Also a path to a file on the remote has to be specified (e.g. `:<filename>` for a file in the user's home directory). The dot can be used as destination to copy the file into the working directory on the local machine. 

--> `virtctl scp -i id_rsa <vm_username>@<vm_name>:<filename> .`